execution/block-partitioner/src/v2/conflicting_txn_tracker.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright © Aptos Foundation
+
+use crate::v2::{OriginalTxnIdx, ShardedTxnIndex2};
+#[cfg(test)]
+use aptos_types::state_store::state_key::StateKey;
+use aptos_types::{
+    block_executor::partitioner::{RoundId, ShardId},
+    transaction::analyzed_transaction::StorageLocation,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::{btree_set::BTreeSet, HashSet};
+
+/// This structure is only used in `V2Partitioner`.
+/// For txns that claimed to access the same storage location,
+/// it caches some metadata about the location and also keeps track of their status (pending or position finalized) throughout the partitioning process.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ConflictingTxnTracker {
+    /// The storage location on which conflicting txns are being tracked by this tracker.
+    pub storage_location: StorageLocation,
+    /// A randomly chosen owner shard of the storage location, for conflict resolution purpose.
+    pub anchor_shard_id: ShardId,
+    /// Txns that (1) read the current storage location and (2) have not been accepted.
+    pending_reads: BTreeSet<OriginalTxnIdx>,
+    /// Txns that (1) write the current storage location and (2) have not been accepted.
+    pending_writes: BTreeSet<OriginalTxnIdx>,
+    /// Txns that write the current storage location.
+    pub writer_set: HashSet<OriginalTxnIdx>,
```

Comment:
What's the difference between `pending_writes` and `writer_set` ?

Reply:
`pending_writes` tracks all writes that haven't been accepted, and is frequently updated during discarding rounds.

`writer_set` is a constant lookup table needed in `add_edges` to check if a txn writes to the current key or not. 

Example:
```
// before any discarding

writer_set == [0, 7, 9]
pending_writes == [0, 7, 9]
accepted == []

// during discarding round 0, mark_txn_accepted(7, round_id=0, shard_id=1)

writer_set == [0, 7, 9]
pending_writes == [0, 9]
accepted == [T7@(0, 1)]

// during discarding round 2, mark_txn_accepted(9, round_id=2, shard_id=3)

writer_set == [0, 7, 9]
pending_writes == [0]
accepted == [T7@(0,1), T9@(2,3)]

// during discarding round 4, mark_txn_accepted(9, round_id=4, shard_id=0)

writer_set == [0, 7, 9]
pending_writes == []
accepted == [T7@(0,1), T9@(2,3), T0@(4,0)]
```

Reply:
This `writer_set` is not needed any more!